### PR TITLE
fix: avoid locking 2 wayland sockets

### LINKED ISSUE #6958: Unable to autostart foot --server
### Regression?

Yes

### System Info and Version

<details>Hyprland, built from branch main at commit 3b6bcd6ddcfebd1fed0694aa2c0bd9753f4d5d46  (presentation-feedback: fix invalid values sent after aq merge).
Date: Sun Jul 21 22:25:07 2024
Tag: v0.41.2-59-g3b6bcd6d, commits: 4945

flags: (if any)

System Information:
System name: Linux
Node name: archbtw
Release: 6.9.10-zen1-1-zen
Version: #1 ZEN SMP PREEMPT_DYNAMIC Thu, 18 Jul 2024 18:05:52 +0000

GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation CometLake-H GT2 [UHD Graphics] [8086:9bc4] (rev 05) (prog-if 00 [VGA controller])
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation TU116M [GeForce GTX 1660 Ti Mobile] [10de:2191] (rev a1) (prog-if 00 [VGA controller])
NVRM version: NVIDIA UNIX x86_64 Kernel Module  555.58.02  Tue Jun 25 01:39:15 UTC 2024

os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo

plugins:

======Config-Start======

Config File: /home/arch-mage/.config/hypr/hyprland.conf: Read Succeeded
#set default monitor
#use 'hyprctl monitors' 
monitor=,preferred,auto,1

#theme
source=$HOME/.config/hypr/mocha.conf

general {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    gaps_in = 5
    gaps_out = 20
    border_size = 1
    col.active_border = rgba($blueAlphaee) rgba($sapphireAlphaee) 45deg
    col.inactive_border = rgba($baseAlphaaa)
    layout = dwindle

    # Please see https://wiki.hyprland.org/Configuring/Tearing/ before you turn this on
    allow_tearing = false
}

group {

    col.border_active = rgba($pinkAlphaee) rgba($tealAlphaee) 45deg
    col.border_inactive = rgba($baseAlphaaa)

groupbar {

    font_size = 11
    text_color= rgba($textAlphaee)
    col.active = rgba($overlay0Alphaee)
    col.inactive = rgba($baseAlphaee)

}

}

cursor {

no_hardware_cursors = false
inactive_timeout = 5

}

decoration {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more

    rounding = 10

    blur {
        enabled = true
        size = 3
        passes = 3
        
        vibrancy = 0.1696
    }

    drop_shadow = true
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba($crustAlphaee)
    dim_inactive = true
    dim_strength = 0.4
    dim_special = 0.3
}

animations {
    enabled = true

    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more

    bezier = myBezier, 0.05, 0.9, 0.1, 1.05

    animation = windows, 1, 7, myBezier
    animation = windowsOut, 1, 7, default, popin 80%
    animation = border, 1, 10, default
    animation = borderangle, 1, 8, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 6, default
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = true # master switch for pseudotiling. Enabling is bound to mainMod + P in the keybinds section below
    preserve_split = true # you probably want this
}

master {
    # See https://wiki.hyprland.org/Configuring/Master-Layout/ for more
    new_status = inherit
}

misc {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    force_default_wallpaper = -1 # Set to 0 or 1 to disable the anime mascot wallpapers
}

#experimental {
#  explicit_sync = yes
#}

layerrule = blur,launcher
layerrule = blur, notifications
layerrule = ignorezero, notifications

#source
source = $HOME/.config/hypr/hyprland.conf.d/env.conf
source = $HOME/.config/hypr/hyprland.conf.d/apps.conf
source = $HOME/.config/hypr/hyprland.conf.d/autostart.conf
source = $HOME/.config/hypr/hyprland.conf.d/windowrules.conf
source = $HOME/.config/hypr/hyprland.conf.d/keybinds.conf
source = $HOME/.config/hypr/hyprland.conf.d/input.conf

Config File: /home/arch-mage/.config/hypr/mocha.conf: Read Succeeded

$rosewater = rgb(f5e0dc)
$rosewaterAlpha = f5e0dc

$flamingo = rgb(f2cdcd)
$flamingoAlpha = f2cdcd

$pink = rgb(f5c2e7)
$pinkAlpha = f5c2e7

$mauve = rgb(cba6f7)
$mauveAlpha = cba6f7

$red = rgb(f38ba8)
$redAlpha = f38ba8

$maroon = rgb(eba0ac)
$maroonAlpha = eba0ac

$peach = rgb(fab387)
$peachAlpha = fab387

$yellow = rgb(f9e2af)
$yellowAlpha = f9e2af

$green = rgb(a6e3a1)
$greenAlpha = a6e3a1

$teal = rgb(94e2d5)
$tealAlpha = 94e2d5

$sky = rgb(89dceb)
$skyAlpha = 89dceb

$sapphire = rgb(74c7ec)
$sapphireAlpha = 74c7ec

$blue = rgb(89b4fa)
$blueAlpha = 89b4fa

$lavender = rgb(b4befe)
$lavenderAlpha = b4befe

$text = rgb(cdd6f4)
$textAlpha = cdd6f4

$subtext1 = rgb(bac2de)
$subtext1Alpha = bac2de

$subtext0 = rgb(a6adc8)
$subtext0Alpha = a6adc8

$overlay2 = rgb(9399b2)
$overlay2Alpha = 9399b2

$overlay1 = rgb(7f849c)
$overlay1Alpha = 7f849c

$overlay0 = rgb(6c7086)
$overlay0Alpha = 6c7086

$surface2 = rgb(585b70)
$surface2Alpha = 585b70

$surface1 = rgb(45475a)
$surface1Alpha = 45475a

$surface0 = rgb(313244)
$surface0Alpha = 313244

$base = rgb(1e1e2e)
$baseAlpha = 1e1e2e

$mantle = rgb(181825)
$mantleAlpha = 181825

$crust = rgb(11111b)
$crustAlpha = 11111b

Config File: /home/arch-mage/.config/hypr/hyprland.conf.d/env.conf: Read Succeeded
#using integrated GPU for rendering
#https://wiki.hyprland.org/Configuring/Multi-GPU/
#lspci | \grep -E 'VGA|3D'
#ls -l /dev/dri/by-path
env = AQ_DRM_DEVICES,/dev/dri/card1

#XDG specs
env = XDG_SESSION_DESKTOP,Hyprland
env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland

#cursor
env = HYPRCURSOR_THEME,catppuccin-mocha-dark
env = HYPRCURSOR_SIZE,24

#other vars
env = GDK_BACKEND,wayland
env = SDL_VIDEODRIVER,wayland
env = CLUTTER_BACKEND,wayland
env = _JAVA_AWT_WM_NONREPARENTING,1
env = QT_QPA_PLATFORM,wayland;xcb
env = QT_QPA_PLATFORMTHEME,qt6ct
env = MOZ_ENABLE_WAYLAND,1
env = EGL_PLATFORM, wayland
env = ELECTRON_OZONE_PLATFORM_HINT,auto 

Config File: /home/arch-mage/.config/hypr/hyprland.conf.d/apps.conf: Read Succeeded
$term = footclient
$filemanager = $term -T nnn  $HOME/.local/bin/nwrap
$menu = fuzzel
$clock = $term -T tty-clock tty-clock -C3 -c -s -n -f "%A %d %B %Y (%Z)"
$editor = $term -T neovim  nvim
$sysmonitor = $term -T htop  htop
$browser = flatpak run org.mozilla.firefox
$screenshot = grim
$screenshot2 = slurp | grim -g - $HOME/Pictures/slurp_screenshot.png
$telegram = flatpak run org.telegram.desktop
$easyeffects = flatpak run com.github.wwmm.easyeffects --gapplication-service
$polkit = /usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
$portals = sleep 1; killall xdg-desktop-portal-hyprland & killall xdg-desktop-portal & sleep 1; /usr/lib/xdg-desktop-portal-hyprland & sleep 2; /usr/lib/xdg-desktop-portal
$mpd_notify = killall -e mpd-notification & sleep 2; mpd-notification -H localhost -p 6600 -m $HOME/Music -s 150 -t 10
$bgreload = killall -e hyprpaper & sleep 1; hyprpaper &
$clipboard = wl-paste --watch cliphist store
$clipboard-copy = cliphist list | fuzzel -d | cliphist decode | wl-copy

Config File: /home/arch-mage/.config/hypr/hyprland.conf.d/autostart.conf: Read Succeeded
exec-once = dbus-update-activation-environment --systemd WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once = systemctl --user import-environment WAYLAND_DISPLAY XDG_CURRENT_DESKTOP
exec-once= $portals &
exec-once = foot --server & $polkit & hypridle & waybar & hyprpaper & $easyeffects & mako & $clipboard & mpd & $mpd_notify &
#exec-once=[workspace 1 silent] $browser & sleep 4; $clock & sleep 1; $filemanager & $telegram &

Config File: /home/arch-mage/.config/hypr/hyprland.conf.d/windowrules.conf: Read Succeeded
# Example windowrule v1
# windowrule = float, ^(kitty)$
# Example windowrule v2
# windowrulev2 = float,class:^(kitty)$,title:^(kitty)$
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more
windowrulev2 = suppressevent maximize, class:.* # You'll probably like this.
windowrulev2=idleinhibit fullscreen, fullscreen:1
windowrulev2=nodim, title:(tty-clock)$ #never dim tty-clock terminal window
windowrulev2=float,class:(ymuse)$ #flatpak Ymuse app
windowrulev2=float,class:(qalculate-gtk)$ #flatpak Qalculate-gtk app

#Appimage - ProtonUp-Qt
windowrulev2=float,class:(net.davidotek.pupgui2)$
windowrulev2=size 600 600,  class:(net.davidotek.pupgui2)$
windowrulev2=center, class:(net.davidotek.pupgui2)$

#bluetoothctl, if launched by clicking on waybar bluetooth module
windowrulev2=float,title:(bluetooth control)$
windowrulev2=size 800 800,title:(bluetooth control)$
windowrulev2=center,title:(bluetooth control)$

#flatpak Planify 
windowrulev2=float, class:(io.github.alainm23.planify)$
windowrulev2=size 1000 800, class:(io.github.alainm23.planify)$
windowrulev2=center, class:(io.github.alainm23.planify)$

#nnn preview-tui plugin
windowrulev2=nofocus, title:(preview-tui)$ 
windowrulev2=nodim, title:(preview-tui)$ 
windowrulev2=group deny, title:(preview-tui)$

#flatpak Mousam
windowrule=size 1200 820, ^(io.github.amit9838.mousam)$ 
windowrulev2=float,class:(io.github.amit9838.mousam)$
windowrulev2=center, class:(io.github.amit9838.mousam)$

#xdg-desktop-portal-gtk
windowrulev2=float, class:(xdg-desktop-portal-gtk)$
windowrulev2=size 800 600,  class:(xdg-desktop-portal-gtk)$

#htop
windowrulev2=float, title:(htop)$
windowrulev2=size 1400 800, title:(htop)$
windowrulev2=center,title:(htop)$

Config File: /home/arch-mage/.config/hypr/hyprland.conf.d/keybinds.conf: Read Succeeded
#mod keys
$mainMod = SUPER
$extraMod = SUPER_SHIFT

#keyboard shortcuts
bind = $mainMod, Return, exec, $term
bind = $mainMod, F, fullscreen, 1
bind = $extraMod, F, fullscreen, 0
bind = $mainMod, Q, killactive,
bind = $extraMod, Q, exec, $HOME/.local/bin/killhypr
bind = $extraMod, R, exec, $bgreload
bind = $mainMod, N, exec, $filemanager
bind = $mainMod, SPACE, togglefloating,
bind = $mainMod, D, exec, $menu
bind = $mainMod, P, pseudo, #also, f4 key # dwindle
bind = $mainMod, X, togglesplit, # dwindle
bind = $mainMod, W, exec, $browser
bind = $mainMod, T, exec, $clock
bind = $mainMod, E, exec, $editor
bind = $mainMod, I, exec, $sysmonitor
bind = $mainMod, Escape, exec, systemctl suspend
bind = $extraMod, Escape, exec, systemctl reboot
bind = $extraMod, A , exec, loginctl lock-session
bind = $mainMod, PRINT, exec, $screenshot
bind = $extraMod, PRINT, exec, $screenshot2
bind = $mainMod, V, exec, $clipboard-copy
bind = $extraMod, T, togglegroup
bind = $extraMod, X, moveoutofgroup
bind = $mainMod, B, changegroupactive, f 
bind = $mainMod, Z, changegroupactive, b
bind = $extraMod, B, movegroupwindow, f
bind = $extraMod, Z, movegroupwindow, b

#mpd-mpc remote control
bind = ,XF86AudioNext, exec, mpc next
bind = ,XF86AudioPrev, exec, mpc prev
bind = ,XF86AudioPlay, exec, mpc toggle

#brightnessctl
bind = ,XF86MonBrightnessUp, exec, brightnessctl set +10%
bind = ,XF86MonBrightnessDown, exec, brightnessctl set 10%-

#wireplumber - volume control
bind = ,XF86AudioRaiseVolume, exec, wpctl set-volume -l 1.0 @DEFAULT_SINK@ 0.05+
bind = ,XF86AudioLowerVolume, exec, wpctl set-volume @DEFAULT_SINK@ 0.05-
bind = ,XF86AudioMute, exec, wpctl set-mute @DEFAULT_SINK@ toggle

# Move focus with mainMod
bind = $mainMod, H, movefocus, l
bind = $mainMod, L, movefocus, r
bind = $mainMod, K, movefocus, u
bind = $mainMod, J, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Example special workspace (scratchpad)
bind = $mainMod, S, togglespecialworkspace, magic
bind = $mainMod SHIFT, S, movetoworkspace, special:magic

# Scroll through existing workspaces with mouse scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Scroll through existing workspaces with keyboard
bind = $mainMod, TAB, workspace, e+1
bind = $extraMod, TAB, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow

#move focused windows with keyboard
bind = $extraMod, H, movewindow, l
bind = $extraMod, L, movewindow, r
bind = $extraMod, K, movewindow, u
bind = $extraMod, J, movewindow, d

#resize windows with keyboard
bind = L_ALT, h, resizeactive, -40 0
bind = L_ALT, l, resizeactive, 40 0
bind = L_ALT, k, resizeactive, 0 -40
bind = L_ALT, j, resizeactive, 0 40

Config File: /home/arch-mage/.config/hypr/hyprland.conf.d/input.conf: Read Succeeded
input {
    kb_layout = us,ru
    kb_variant =
    kb_model =
    kb_options = grp:alt_shift_toggle
    kb_rules =

    follow_mouse = 1
    float_switch_override_focus = 0

touchpad {
    natural_scroll = false
    clickfinger_behavior = 1
    tap-and-drag = true
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}

gestures {
    # See https://wiki.hyprland.org/Configuring/Variables/ for more
    workspace_swipe = true
    workspace_swipe_fingers = 3
    workspace_swipe_min_fingers = true
}

#per-device configs
#device {
#    name = epic-mouse-v1
#    sensitivity = -0.5
#}

======Config-End========

</details>

### Description

foot --server fails to start automatically, if added to autostart but works, if started manually. Attempting to start footclient via foot (no server) instances gives this output

```
warn: client.c:381: /run/user/1000/foot-wayland-2.sock: failed to connect, will now try /run/user/1000/foot.sock
warn: client.c:388: /run/user/1000/foot.sock: failed to connect, will now try /tmp/foot.sock
 err: client.c:394: failed to connect (is 'foot --server' running?): No such file or directory
```

Strange enough, but, it's possible to autostart footclient tui programmes (e.g footclient tty-clock) from the hyprland config, after autostarting foot --server, itself, bit it's imposible to start footclient, afterwards

### How to reproduce

add exec-once = foot --server to config and bind footclient to a keybind or try to start it from terminal. Additionally, try to add some tui programm (e.g footclient tty-clock) to autostart.

### Crash reports, logs, images, videos

_No response_

### PR DESCRIPTION
**Pull request by diniamo**
#### Describe your PR, what does it fix/add?

Fixes #6958

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

Not sure whether the signal call in startCompositor should be moved as well.

#### Is it ready for merging, or does it need work?

Yes, it works.

### CODE REVIEW

**Review by vaxerski — DISMISSED** [review]

**Comment by vaxerski** [comment]
needs a clang-format

File: src/Compositor.cpp [hunk 1]
```diff
@@ -206,8 +206,7 @@ void CCompositor::setRandomSplash() {
 
 static std::vector<SP<Aquamarine::IOutput>> pendingOutputs;
 
-//
-void CCompositor::initServer() {
+void                                        CCompositor::initServer(std::string socketName, int socketFd) {
```
Comment (vaxerski) [inline src/Compositor.cpp:209]: and this is why the `//` was above this :) please add it back

Comment (diniamo) [reply src/Compositor.cpp:209]: ??????????????????????

**Review by vaxerski — APPROVED** [review]
thanks!

File: src/Compositor.cpp (hunk 1 above)
Comment (vaxerski) [reply src/Compositor.cpp]: :)

**Pull request merged**